astropy/coordinates/earth.py

```diff
@@ -848,3 +864,84 @@ def _to_value(self, unit, equivalencies=[]):
             equivalencies = self._equivalencies
         new_array = self.unit.to(unit, array_view, equivalencies=equivalencies)
         return new_array.view(self.dtype).reshape(self.shape)
+
+
+class BaseGeodeticRepresentation(BaseRepresentation):
+    """
+    Representation of points in 3D geodetic coordinates.
+
+    Parameters
+    ----------
+    lon, lat : `~astropy.units.Quantity`
+        The longitude and latitude of the point(s), in angular units. The
+        latitude should be between -90 and 90 degrees, and the longitude will
+        be wrapped to an angle between 0 and 360 degrees. These can also be
+        instances of `~astropy.coordinates.Angle`,
+        `~astropy.coordinates.Longitude`, or `~astropy.coordinates.Latitude`.
+
+    height : `~astropy.units.Quantity`
+        The height to the point(s).
+
+    copy : bool, optional
+        If `True` (default), arrays will be copied. If `False`, arrays will
+        be references, though possibly broadcast to ensure matching shapes.
+
+    """
+
+    attr_classes = {'lon': Longitude,
+                    'lat': Latitude,
+                    'height': u.Quantity}
+
+    def __init__(self, lon, lat=None, height=None, copy=True):
+        if height is None and not isinstance(lon, self.__class__):
+            height = Distance(0, 'm')
+        super().__init__(lon, lat, height, copy=copy)
+
+    def to_cartesian(self):
+        """
+        Converts WGS84 geodetic coordinates to 3D rectangular (geocentric)
+        cartesian coordiantes.
+        """
+        xyz = erfa.gd2gc(getattr(erfa, self._ellipsoid),
+                         self.lon.to_value(u.radian),
+                         self.lat.to_value(u.radian),
+                         self.height.to_value(u.m))
+        return CartesianRepresentation(xyz, unit=u.m, xyz_axis=-1, copy=False)
+
+    @classmethod
+    def from_cartesian(cls, cart):
+        """
+        Converts 3D rectangular cartesian coordinates (assumed geocentric) to
+        WGS84 geodetic coordinates.
+        """
+        lon, lat, height = erfa.gc2gd(getattr(erfa, cls._ellipsoid),
+                                      cart.get_xyz(xyz_axis=-1).to_value(u.m))
+        return cls(
+            Longitude(lon << u.radian, u.degree,
+                      wrap_angle=180. << u.degree, copy=False),
+            Latitude(lat << u.radian, u.degree, copy=False),
+            u.Quantity(height << u.meter, copy=False))
+
+
+class WGS84GeodeticRepresentation(BaseGeodeticRepresentation):
+    """
```

Comment:
One disadvantage of setting this docstring is that you lose the nice information you added in the `Base...`.  For a possible solution, see https://docs.astropy.org/en/latest/api/astropy.utils.decorators.format_doc.html#astropy.utils.decorators.format_doc

Reply:
I ended up removing the docstrings, after all the name should be explicit enough.

However, I did a local `tox -e build_docs` and I didn't see these classes anywhere in the docs, so I added them to `__all__` in `earth.py`.

Reply:
I was wrong, removing the docstrings leaves the class documentation empty:

![Screenshot_2020-12-01 WGS84GeodeticRepresentation — Astropy v4 3 dev322+g7f34535cc d20201201](https://user-images.githubusercontent.com/316517/100720080-c57bea00-33bd-11eb-83a8-ca18579e170e.png)

Reply:
`@format_doc` proved to be the best solution :+1: